extensions/amp-accordion/1.0/accordion.js

```diff
@@ -260,9 +278,14 @@ export function AccordionSection({
 
   useLayoutEffect(() => {
     if (registerSection) {
-      return registerSection(id, defaultExpanded);
+      return registerSection(id, defaultExpanded, onExpand, onCollapse);
```

Comment:
Can we instead just trigger events from here? Do we have to propagate the handlers up to the accordion?

Reply:
So, I got the point about the imperative APIs. In this case, IMHO, this is good. But there's one important nuance: `onExpand` and `onCollapse` (and many other event callbacks) are often recreated. We cannot assume that they will stay invariant through the life of the section. And we don't want to keep re-registering the section since no actual logic behind it changes. What we should do instead is to create a ref right above the `useLayoutEffect` with the `{onExpand, onCollapse}` and pass this ref instead.

E.g.

```
const sectionPropsRef = useRef();
sectionPropsRef.current = {onExpand, onCollapse};
useLayoutEffect(() => {
  return registerSection(id, defaultExpanded, sectionpropsRef);
}, [registerSection, id, defaultExpanded]);
```

And as a tiny benefit, handling this in the `registerSection()` method in `Accordion` will be easier:

```
registerSection = useCallback((id, defaultExpanded, sectionPropsRef) => {
  eventMap.current[id] = sectionPropsRef;
}
```

The only nuance is that you need to be careful with types here and declare a `typedef` for the `sectionPropsRef`.

And, I also named it `sectionPropsRef` so that, in the future, we can propagate other props too if need arises, not just "events". But if you think it's unlikely, you can name accordingly.

Reply:
If we only keep one `onExpandStateChanged`, then we can simply make `const onExpandStateChangefRef = useRef(); onExpandStateChangefRef.current = onExpandStateChanged`

Reply:
Thanks for the review here!  I've decided to go with a single `onExpandStateChange` event function.  And also went with `sectionPropsRef` in case we want to add other props in the future.